### DISCUSSION TITLE
LibPandaDNA Support

### DISCUSSION THREAD

**Discussion by furtabs** [OpenToontownTools/TTOpenLevelEditor#38] [General]
## Description

Reupload of #29 since I derped the description.

I am requesting the ability for the level editor to completely bypass non open source components and the ability to use LibPandaDNA to launch the editor similar to the scope of Chris's Project [Toon Express](https://github.com/chrisd149/Toon-Express-Level-Editor). 

My biggest pet pieve is this is the **Open** Level Editor and LibDisney is **not** open source, but is Visible Source. I believe all open source components should be used if possible. With the codebase of LibDisney being leaked / shared to the public maybe it could be used to understand the code better to help finish the issue on LibPandaDNA's side.

**Comment by kris-diaz** [comment]
The main issue I ran into using Libpandadna to launch the editor was the lack of a couple of components (mainly DNASignText.)  I believe the missing components have since been added to the [original libpandadna repo](https://github.com/loblao/libpandadna) and could be rigged to work with the level editor.  I'm not sure how much effort/refracturing it would take, but from my own attempt at it, it looks like it could work.

**Comment by drewc5131** [comment]
Again, this sounds like a complete waste of time considering that the [original disney code (but modernized to work with panda 1.10+ and python 3)](https://github.com/OpenToontownTools/libtoontown) is just as easy to compile as libpandadna, and libpandadna is being phased out in favor of original disney code (e.g. toontown offline), and ttoffline is the only really relevant game that used libpandadna. (i exclude clash due to it being an old heavily modified python version of libpandadna that is internally very different)

I'll leave this open for discussion but the reality is, libpandadna would be the project that needs updating for compatibility in order to add all missing features that are required, since libpandadna is explicitly coded to be a drag and drop replacement for libtoontown, so all of the functions and components should line up now, but if anything else is missing, thats on them, not the editor.
